File: SerpentineApi/SerpentineApi/Responses/ChannelMemberResponse.cs
namespace SerpentineApi.Responses;

public class ChannelMemberResponse : BaseResponse
{
    public Ulid ChannelId { get; set; }
    public Ulid UserId { get; set; }
    public bool IsSilenced { get; set; }
    public bool IsArchived { get; set; }
    public bool IsOwner { get; set; }

    public DateTime LastAccess { get; set; } = DateTime.Now;

    public string UserProfilePictureUrl { get; set; } = "";

    public string UserUsername { get; set; } = "";

    public string UserName { get; set; } = "";

    public ChannelMemberRoleResponse Role { get; set; } = new();
    
    
}

File: serpentineclient/src/models/responses/channel-member-response.ts
import { BaseResponse } from "./base-response";

export class ChannelMemberResponse extends BaseResponse {
    
  channelId: string = "";
  userId: string = "";
  isSilenced: boolean = false;
  isArchived: boolean = false;
  isOwner: boolean = false;
  isOverage: boolean = false;
  lastAccess: Date = new Date();
  userProfilePictureUrl : string = "";
  userUsername : string = "";
  userName : string = "";
  isAdmin: boolean = false;
  isActive: boolean = false;


  constructor(init?: Partial<ChannelMemberResponse>) {
    super(init); 
    Object.assign(this, init);
  }
}